src/modules/auth/tests/login.e2e-spec.ts

```diff
@@ -0,0 +1,84 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { INestApplication } from '@nestjs/common';
+import * as request from 'supertest';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { User } from '../../../entities/user.entity';
+import { JwtService } from '@nestjs/jwt';
+import * as bcrypt from 'bcryptjs';
+import { AppModule } from '../../../app.module';
+
+describe('AuthController (e2e)', () => {
+  let app: INestApplication;
+  let mockUserRepository;
+  let mockJwtService;
+
+  const mockUser = {
+    id: '1',
+    email: 'test@example.com',
+    password: 'hashedPassword',
+  };
+
+  beforeAll(async () => {
+    mockUserRepository = {
+      findOneBy: jest.fn(),
+    };
+
+    mockJwtService = {
+      sign: jest.fn().mockReturnValue('fake-jwt-token'),
+    };
+
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule],
+    })
+      .overrideProvider(getRepositoryToken(User))
+      .useValue(mockUserRepository)
+      .overrideProvider(JwtService)
+      .useValue(mockJwtService)
+      .compile();
+
+    app = moduleFixture.createNestApplication();
+    app.setGlobalPrefix('api/v1');
+    await app.init();
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
+
+  it('/api/v1/auth/login (POST) - success', async () => {
```

Comment:
write proper test statements

"it("should login a user", blah blah"

Reply:
Resolved